hack/tools/runtime-openapi-gen/vendored_openapi.go

```diff
@@ -0,0 +1,455 @@
+package main
+
+import (
+	v1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/kube-openapi/pkg/common"
+	"k8s.io/kube-openapi/pkg/validation/spec"
+)
+
+// GetOpenAPIDefinitions returns OpenAPIDefinitions for external types.
+// We need them when we want to generate OpenAPI for a type embedding one of those external types.
+// Those OpenAPIDefinitions are all vendored from: https://github.com/kubernetes/kubernetes/blob/master/pkg/generated/openapi/zz_generated.openapi.go
+// Prior art: https://github.com/kubernetes-sigs/controller-tools/blob/master/pkg/crd/known_types.go.
+func GetOpenAPIDefinitions(ref common.ReferenceCallback) map[string]common.OpenAPIDefinition {
```

Comment:
TBD if we want to move this into the catalog.

I'm fine with both options. Opinions?

Reply:
On the other side the chance that anyone needs the same or a subset of our vendored OpenAPI Definitions is probably very small.